packages/flutter/lib/src/widgets/image.dart

```diff
@@ -108,6 +108,80 @@ Future<void> precacheImage(
   return completer.future;
 }
 
+/// Signature used by [Image.frameBuilder] to control the widget that will be
+/// used when an [Image] is built.
+///
+/// The `child` argument contains the default widget that would be rendered if
+/// no [Image.frameBuilder] had been specified. In other words, returning
+/// `child` directly from this method will yield the same result as this method
+/// having not been consulted. This argument is guaranteed to be non-null.
+/// Typically, this builder will wrap the `child` widget in some way and return
+/// the wrapped widget.
+///
+/// The `frame` argument specifies the index of the current image frame being
+/// rendered. It will be null before the first image frame is ready, and zero
+/// for the first image frame. For single-frame images, it will never be greater
+/// than zero. For multi-frame images (such as animated GIFs), it will increase
+/// by one every time a new image frame is shown (including when the image
+/// animates in a loop).
+///
+/// The `wasSynchronouslyLoaded` argument specifies whether the image was
+/// available synchronously (immediately). If this is true, then the image was
```

Comment:
your parenthetical is redundant with the following sentence. Maybe `(on the same frame as the `Image` widget itself was created)` or some such?

Reply:
Done